src/icp_canister_backend/tests/rewards_tests.rs

```diff
@@ -159,4 +199,649 @@ fn test_reward_distribution_middle_and_final() {
+    let first_reward = Nat::from(50_000_000u64); // 0.5 BTC
+    let second_reward = Nat::from(30_000_000u64); // 0.3 BTC
+    let third_reward = Nat::from(20_000_000u64); // 0.2 BTC
+    let total_expected_rewards =
+        first_reward.clone() + second_reward.clone() + third_reward.clone(); // 1 BTC total
+
+    let stakable_episode = get_stakable_episode(&pic, canister_id, 7);
+    create_deposit(
+        &pic,
+        canister_id,
+        ledger_id,
+        user,
+        deposit_amount.clone(),
+        stakable_episode,
+    );
+
+    let first_reward_time = pic.get_time().as_nanos_since_unix_epoch() / 1_000_000_000;
+
+    // Add all three rewards
+    reward_pool(&pic, canister_id, ledger_id, user, first_reward.clone())
+        .expect("First reward pool should succeed");
+
+    advance_time(&pic, EPISODE_DURATION / 4);
+    let second_reward_time = pic.get_time().as_nanos_since_unix_epoch() / 1_000_000_000;
+    reward_pool(&pic, canister_id, ledger_id, user, second_reward.clone())
+        .expect("Second reward pool should succeed");
+
+    advance_time(&pic, EPISODE_DURATION / 4);
+    let third_reward_time = pic.get_time().as_nanos_since_unix_epoch() / 1_000_000_000;
+    reward_pool(&pic, canister_id, ledger_id, user, third_reward.clone())
+        .expect("Third reward pool should succeed");
+
+    let first_end =
+        ((first_reward_time + EPISODE_DURATION * 12) / EPISODE_DURATION + 1) * EPISODE_DURATION;
+    let second_end =
+        ((second_reward_time + EPISODE_DURATION * 12) / EPISODE_DURATION + 1) * EPISODE_DURATION;
+    let third_end =
+        ((third_reward_time + EPISODE_DURATION * 12) / EPISODE_DURATION + 1) * EPISODE_DURATION;
+
+    let latest_end = first_end.max(second_end).max(third_end);
+    let current_time = pic.get_time().as_nanos_since_unix_epoch() / 1_000_000_000;
+
+    advance_time(&pic, latest_end - current_time);
+
+    let final_rewards = pic
+        .query_call(
+            canister_id,
+            user,
+            "get_deposits_rewards",
+            encode_args((vec![0u64],)).unwrap(),
+        )
+        .map(|r| decode_one::<Nat>(&r).unwrap())
+        .expect("Failed to get final rewards");
+
+    assert_with_error!(
+        &final_rewards,
+        &total_expected_rewards,
+        &ALLOWED_ERROR,
+        "Multiple reward additions final total"
+    );
+}
+
+#[test]
+fn test_reward_distribution_between_large_and_small_deposits() {
+    let (pic, canister_id, ledger_id) = setup();
+    let large_depositor = Principal::from_text("rdmx6-jaaaa-aaaaa-aaadq-cai").unwrap();
+    let small_depositor = Principal::from_text("rrkah-fqaaa-aaaaa-aaaaq-cai").unwrap();
+
+    let large_deposit = Nat::from(1_000_000_000u64); // 10 BTC ($1M USD at $100k/BTC)
+    let small_deposit = icp_canister_backend::MINIMUM_DEPOSIT_AMOUNT.clone() + Nat::from(1u64); // Just above minimum
+
+    let reward_amount = Nat::from(10_000_000u64); // $1000 USD in BTC (0.01 BTC at $100k/BTC)
```

Comment:
Here is the right place for "$1000 USD in BTC (0.01 BTC at $100k/BTC)" comment
Make 1$ reward, if it fails, research why, and with what values.